File: crates/yauth/src/domain/passkey.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WebauthnCredential {
    pub id: Uuid,
    pub user_id: Uuid,
    pub name: String,
    pub aaguid: Option<String>,
    pub device_name: Option<String>,
    pub credential: serde_json::Value,
    pub created_at: NaiveDateTime,
    pub last_used_at: Option<NaiveDateTime>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NewWebauthnCredential {
    pub id: Uuid,
    pub user_id: Uuid,
    pub name: String,
    pub aaguid: Option<String>,
    pub device_name: Option<String>,
    pub credential: serde_json::Value,
    pub created_at: NaiveDateTime,
}

File: domain/passkey.go
package domain

import (
	"encoding/json"
	"time"
)

// WebauthnCredential is a registered WebAuthn / passkey credential.
type WebauthnCredential struct {
	ID         string
	UserID     string
	Name       string
	AAGUID     *string
	DeviceName *string
	Credential json.RawMessage
	CreatedAt  time.Time
	LastUsedAt *time.Time
}

// NewWebauthnCredential is the input for registering a WebAuthn credential.
type NewWebauthnCredential struct {
	ID         string
	UserID     string
	Name       string
	AAGUID     *string
	DeviceName *string
	Credential json.RawMessage
	CreatedAt  time.Time
}
